server/world/world.go

```diff
@@ -777,6 +789,41 @@ func (w *World) Entities() []Entity {
 	return m
 }
 
+// Sleepers returns a list of all sleeping entities currently added to the World.
+func (w *World) Sleepers() []Sleeper {
+	ent := w.Entities()
+	sleepers := make([]Sleeper, 0, len(ent))
```

Comment:
This will usually allocate a slice that is way too big, because hardly any entities are sleepers. I would reduce this significantly, maybe `len(ent)/40` or something like that.